### ISSUE TITLE
Rule to support deprecations: tmpdir->tmp_path && tmpdir_factory->tmp_path_factory

### ISSUE THREAD

**Issue by webknjaz** [m-burst/flake8-pytest-style#23]
# Rule request

## Description

Basically, whenever a deprecated fixture is encountered, the linter should warn about this and suggest a better replacement as per $sbj.

## Rationale

[The pytest fixtures list] suggests that there's better alternatives for `tmpdir` and `tmpdir_factory` that return old-style shims:
> `tmp_path`
>    Provide a `pathlib.Path` object to a temporary directory which is unique to each test function.
> `tmp_path_factory`
>    Make session-scoped temporary directories and return `pathlib.Path` objects.
> `tmpdir`
>    Provide a `py.path.local` object to a temporary directory which is unique to each test function; **replaced by `tmp_path`**.
> `tmpdir_factory`
>    Make session-scoped temporary directories and return `py.path.local` objects; **replaced by `tmp_path_factory`**.

I saw this in a few other places in the docs but it's not emphasized enough. I often don't remember which one to use and add `tmpdir` expecting `pathlib(2).Path` objects there when I should use `tmp_path` instead to achieve this. Every time I have to go to docs to find out which one is old.
It's PITA and a new rule could be very helpful here, plus it could be a framework for any future deprecations that may happen later.

[The pytest fixtures list]: https://docs.pytest.org/en/latest/fixture.html

**Comment by m-burst** [comment]
Another example of a deprecated fixture is `mock` -> `mocker` from `pytest-mock`.

**Comment by webknjaz** [comment]
Right. Also: `monkeypatch`, especially where mock objects are not used, but it's harder to detect.
